pkg/controllers/tunnel/tunnel.go

```diff
@@ -27,20 +27,39 @@ import (
 	v1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
 	"github.com/submariner-io/submariner/pkg/cableengine"
 	"k8s.io/apimachinery/pkg/runtime"
+	k8snet "k8s.io/utils/net"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 )
 
 type controller struct {
-	engine cableengine.Engine
+	engine          cableengine.Engine
+	localIPFamilies [2]k8snet.IPFamily
 }
 
 var logger = log.Logger{Logger: logf.Log.WithName("Tunnel")}
 
+func findCommonIPFamilies(local, remote [2]k8snet.IPFamily) []k8snet.IPFamily {
+	common := []k8snet.IPFamily{}
+
+	for _, lf := range local {
+		for _, rf := range remote {
+			if lf == rf {
+				common = append(common, lf)
+				break
+			}
+		}
+	}
+
+	return common
```

Comment:
You can simplify to (import "github.com/submariner-io/admiral/pkg/slices"):

```suggestion
	return slices.Intersect(local, remote, func(e k8snet.IPFamily) k8snet.IPFamily {
		return e
	})
```

Reply:
Addressed by commit https://github.com/submariner-io/submariner/pull/3311/commits/6d5338cf9b059031032fbc1df8e302db930d8057.